sktime/transformations/series/difference.py

```diff
@@ -0,0 +1,317 @@
+        lags = check_array(lags, ensure_2d=False)
+        if (lags <= 0).any():
+            raise ValueError(non_positive_msg)
+    else:
+        raise TypeError(msg)
+
+    return lags
+
+
+def _inverse_diff(series, lag):
+    for i in range(lag):
+        series.iloc[i::lag] = series.iloc[i::lag].cumsum()
+
+    return series
+
+
+class Differencer(_SeriesToSeriesTransformer):
+    """Apply iterative differences to a timeseries.
+
+    Difference transformations are applied at the specified lags in the order
+    provided. For example, using lags=[1, 12] corresponds to applying a
+    standard first difference, then differencing the first-differenced series
+    at lag 12 (in the event the input data has a monthly periodicity, this
+    would equate to a first difference followed by a seasonal difference).
+
+    The transformation works for univariate and multivariate timeseries. However,
+    the multivariate case applies the same differencing to every series.
+
+    Parameters
+    ----------
+    lags : int or array-like, default = 1
+        The lags used to difference the data.
+        If a single `int` value is
+
+    remove_missing : bool, default = True
+        Whether the differencer should remove the initial observations that
+        contain missing values as a result of the differencing operation(s).
+
+    Attributes
+    ----------
+    lags : np.ndarray
+        Lags used to perform the differencing of the input series.
+
+    remove_missing : bool
+        Stores whether the Differencer removes initial observations that contain
+        missing values as a result of the differencing operation(s).
+
+    Example
+    -------
+    >>> from sktime.transformations.series.difference import Differencer
+    >>> from sktime.datasets import load_airline
+    >>> y = load_airline()
+    >>> transformer = Differencer(lags=[1, 12])
+    >>> y_hat = transformer.fit_transform(y)
+    """
+
+    _tags = {
+        "fit-in-transform": False,
+        "transform-returns-same-time-index": False,
+        "univariate-only": False,
+    }
+
+    def __init__(self, lags=1, remove_missing=True):
+        self.lags = lags
+        self.remove_missing = remove_missing
+        self._Z = None
+        self._cumulative_lags = None
+        self._prior_cum_lags = None
+        self._prior_lags = None
+        super(Differencer, self).__init__()
+
+    def _check_inverse_transform_index(self, Z):
+        """Check fitted series contains indices needed in inverse_transform."""
+        first_idx = Z.index.min()
+        orig_first_idx, orig_last_idx = self._Z.index.min(), self._Z.index.max()
+
+        is_contained_by_fitted_z = False
+        is_future = False
+
+        if first_idx < orig_first_idx:
+            msg = [
+                "Some indices of `Z` are prior to timeseries used in `fit`.",
+                "Reconstruction via `inverse_transform` is not possible.",
+            ]
+            raise ValueError(" ".join(msg))
```

Comment:
What's the different compared to passing the `msg` directly?
```suggestion
            raise ValueError(msg)
```